src/application-base/ApplicationBase.jsx

```diff
@@ -77,10 +44,18 @@ const propTypes = {
 };
 
 const ApplicationBase = ({
-  locale, customTranslatedMessages, translationsLoadingPlaceholder, themeName, fitToParentIsDisabled, children, unloadPromptIsDisabled,
+  locale, themeName, children, unloadPromptIsDisabled,
 }) => {
   const registeredPromptsRef = useRef();
 
+  const { localeOverride } = useTestOverrides(); // Allows us to test deployed applications in different locales.
+
+  const theme = useMemo(() => ({
+    // If the theme class name is undefined or an empty string, that indicates we have the root theme and should apply the root theme name.
+    name: themeName || rootThemeName,
```

Comment:
Should we expect the default theme name to come from the theme configuration? The fallback on this is a little weird since this could come from three places.

Reply:
`rootThemeName` is either the 'theme' field from theme config, or 'terra-default-theme' if it's not set. When we first setup the theme config, `theme` was and still is optional, so this gives a name to the default theme.